server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -186,13 +234,27 @@ static final class Fields {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field(Fields.COUNT, indexCount);
-        shards.toXContent(builder, params);
-        docs.toXContent(builder, params);
-        store.toXContent(builder, params);
-        fieldData.toXContent(builder, params);
-        queryCache.toXContent(builder, params);
-        completion.toXContent(builder, params);
-        segments.toXContent(builder, params);
+        if (shards != null) {
+            shards.toXContent(builder, params);
```

Comment:
Instead of adding null checks at all places, why not just add a parameter in the method itself to do nothing if null is received?

Reply:
Moving the null checks inside the method will cause null pointer exceptions when the objects are null on calling toXContent methods.

Reply:
@SwethaGuptha Did you get a chance to check this?